build/mongodl.py

```diff
@@ -0,0 +1,665 @@
+                mem.isdir(),
+                lambda: tf.extractfile(mem),
+                mem.mode,
+                test=test,
+            )
+    return n_extracted
+
+
+def _expand_zip(ar, dest, pattern, strip_components, test):
+    'Expand a .zip archive.'
+    n_extracted = 0
+    with zipfile.ZipFile(ar, 'r') as zf:
+        for item in zf.infolist():
+            n_extracted += _maybe_extract_member(
+                dest,
+                item.filename,
+                pattern,
+                strip_components,
+                item.is_dir(),
+                lambda: zf.open(item, 'r'),
+                0o655,
+                test=test,
+            )
+    return n_extracted
+
+
+def _maybe_extract_member(out, relpath, pattern, strip, is_dir, opener,
+                          modebits, test):
+    """
+    Try to extract an archive member according to the given argumetns.
+
+    :return: Zero if the file was excluded by filters, one otherwise.
+    """
+    relpath = PurePath(relpath)
+    print('  │ {:┄<65} │'.format(str(relpath) + ' '), end='')
+    if len(relpath.parts) <= strip:
+        # Not enough path components
+        print(' (Excluded by --strip-components)')
+        return 0
+    if not _test_pattern(relpath, pattern):
+        # Doesn't match our pattern
+        print(' (excluded by pattern)')
+        return 0
+    stripped = pathjoin(relpath.parts[strip:])
+    dest = Path(out) / stripped
+    print('\n    -> [{}]'.format(dest))
+    if test:
+        # We are running in test-only mode: Do not do anything
+        return 1
+    if is_dir:
+        dest.mkdir(exist_ok=True, parents=True)
+        return 1
+    with opener() as infile:
+        dest.parent.mkdir(exist_ok=True, parents=True)
+        with dest.open('wb') as outfile:
+            shutil.copyfileobj(infile, outfile)
+        os.chmod(str(dest), modebits)
+    return 1
+
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__)
+    grp = parser.add_argument_group('List arguments')
+    grp.add_argument('--list',
+                     action='store_true',
+                     help='List available components, targets, editions, and '
+                     'architectures. Donwload arguments will act as filters.')
+    dl_grp = parser.add_argument_group(
+        'Download arguments',
+        description='Select what to download and extract. '
+        'Non-required arguments will be inferred '
+        'based on the host system.')
+    dl_grp.add_argument('--target',
+                        '-T',
+                        help='The target platform for which to download. '
+                        'Use "--list" to list available targets.')
+    dl_grp.add_argument('--arch',
+                        '-A',
+                        help='The architecture for which to download')
+    dl_grp.add_argument(
+        '--edition',
+        '-E',
+        help='The edition of the product to download (Default is "targeted"). '
```

Comment:
I am not sure what "targeted" means. Since this script is intended for DBX engineers, I suggest setting a default of "enterprise".

Reply:
I'm not sure what it means either. I just chose it as an arbitrary default. I'm finding that "`enterprise`" will be better since one needs it for `csfle` on `mongocryptd` anyway.